controllers/ibmpowervsmachine_controller.go

```diff
@@ -289,12 +306,10 @@ func (r *IBMPowerVSMachineReconciler) reconcileNormal(machineScope *scope.PowerV
 		machineScope.Info("Unable to update the LoadBalancer, Machine internal IP not yet set", "machine name", machineScope.IBMPowerVSMachine.Name)
 		return ctrl.Result{}, nil
 	}
-	poolMember, err := machineScope.CreateVPCLoadBalancerPoolMember()
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed CreateVPCLoadBalancerPoolMember %s: %w", machineScope.IBMPowerVSMachine.Name, err)
-	}
-	if poolMember != nil && *poolMember.ProvisioningStatus != string(infrav1beta2.VPCLoadBalancerStateActive) {
-		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
+
+	if poolMemberReconcileResult, err := r.handleLoadBalancerPoolMemberConfiguration(machineScope); err != nil || poolMemberReconcileResult.RequeueAfter > 0 {
```

Comment:
The handleLoadBalancerPoolMemberConfiguration was introduced more on the lines of a helper func that handles 1. controlPlaneMachine check, 2. machineScope.CreateVPCLoadBalancerPoolMember and 3. check if poolmember provisioning status is not active. So may not be apt to add as a single responsibility function that is similar to the existing machineScope methods.

To your second question, we could go with something like this 
`if util.IsControlPlaneMachine(machineScope.Machine) {
		return r.handleLoadBalancerPoolMemberConfiguration(machineScope)
}` 

because adding all the conditions on the parent will cause the current cyclomatic complexity to cross threshold. Let me know your opinion

Reply:
Got it thanks.

Reply:
Updated as discussed